### ISSUE TITLE
common conon data for multiple developer

### ISSUE THREAD

**Issue by mdzahid989** [conan-io/docs#3163]
Hi,

Presently , when i do conan install - All the dependencies are downloaded in CONAN_USER_HOME location.

If there are 100 developers - that consume same dependencies - it takes 100x disk space.
Is there any possibility in conan - where multiple developer share dependencies in common location ?

**Comment by memsharded** [comment]
Hi @mdzahid989 

Thanks for your question.

The Conan local package cache is not concurrent, so it cannot be shared. It is intended to be local to the developer computer and private to the developer. 
But this shouldn't be a problem, also the compilers, cmake, the IDE, and many other tools and other things will be installed in the 100 developer machines taking 100x space, isn't it? Is this a problem in your case? Disk nowadays is very cheap, and developers have plenty of it in their machines.

Take into account that having a shared cache has other problems:

- All developers have access to all  other packages of all other developers and builds. In many cases there are policies where not all developers have the same permissions, and shouldn't be able to read some packages
- Users could do destructive things to other developers: ``conan remove * -c`` removes all packages in the cache. If developers do have write permissions (because they should have write permissions, otherwise they cannot ``conan install`` dependencies, or they cannot ``conan create`` their own packages), then they can very easily wipe the cache for everyone
- Unintended changes breaks other developer builds: If some developers is experimenting with an improvement for the ``zlib/1.2.13`` package, and do a ``conan create`` for it, and creates a binary for that package in a shared cache that crashes, then suddenly all developers start to use that binary in their builds, and that would be breaking all developers builds.

In other words, having a shared cache for all developers is like having a single source Git repository, with a single branch, that everyone can commit to that branch at any time. So using per-developer local caches is the way to go.

**Comment by mdzahid989** [comment]
Thanks very much @memsharded  - for Detailed reply.
